File: LeetCode/DataStruct/Python/Leetcode1725-可以形成最大正方形的矩形数目.py
class Solution:
    def countGoodRectangles(self, rectangles: List[List[int]]) -> int:
        st = defaultdict(int)
        for edge in rectangles:
            st[min(edge[0],edge[1])] += 1
        mx,cnt = -1,0
        for k,v in st.items():
            if k > mx:
                mx,cnt = k,v
        return cnt

File: LeetCode/DataStruct/JS/Leetcode1725-可以形成最大正方形的矩形数目.js
var countGoodRectangles = function(rectangles) {
    let mx = -1,cnt = 0;
    for(const edge of rectangles) {
        let key = Math.min(edge[0],edge[1]);
        if(mx == key) cnt++;
        else if(mx < key) {
            cnt = 1;
            mx = key;
        }
    }
    return cnt;
};